### PR TITLE
Add scroll functionality for mobile

### PR DESCRIPTION
**Pull request by anishathalye**
This patch adds functionality to be able to scroll on mobile (tested with an iOS device).

### CODE REVIEW

File: src/Viewport.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ export class Viewport {
     this.terminal.on('scroll', this.syncScrollArea.bind(this));
     this.terminal.on('resize', this.syncScrollArea.bind(this));
     this.viewportElement.addEventListener('scroll', this.onScroll.bind(this));
+    this.touchY = null;
```
Comment (Tyriar) [inline src/Viewport.ts]: You'll need to declare this as a private member above to remove the compile warning. A better name might also be `lastTouchY`:

```ts
private lastTouchY: number;
```

File: src/xterm.js [hunk 2]
```diff
@@ -1086,6 +1086,18 @@ Terminal.prototype.bindMouse = function() {
     self.viewport.onWheel(ev);
     return self.cancel(ev);
   });
+
+  on(el, 'touchstart', function(ev) {
+    if (self.mouseEvents) return;
+    self.viewport.onTouchStart(ev);
+    return self.cancel(ev);
+  }
```
Comment (Tyriar) [inline src/xterm.js]: Missing a `);` at the end

File: src/xterm.js [hunk 3]
```diff
@@ -1086,6 +1086,18 @@ Terminal.prototype.bindMouse = function() {
...
+
+  on(el, 'touchmove', function(ev) {
+    if (self.mouseEvents) return;
+    self.viewport.onTouchMove(ev);
+    return self.cancel(ev);
+  }
```
Comment (Tyriar) [inline src/xterm.js]: Missing a `);` at the end

**Comment by anishathalye** [comment]
Thanks for the feedback! I rebased onto the current master branch and also made the changes you asked for.

(for some reason, GitHub doesn't seem to recognize my `commit --amend` as a change...)

Let me know if there's anything else I should change.

**Comment by Tyriar** [comment]
@anishathalye yeah with GitHub you typically add individual "fix" commits after a PR is made as amending a commit that has already been commented on can turn out a bit weird in GH. The repo owners can squash the commits if they want when it's merged.

**Comment by tbodt** [comment]
The problem with this is it doesn't support the iOS scroll momentum thing. I've gotten that to work by removing these listeners, using z-index to put the viewport div above the rows div, and putting `-webkit-overflow-scrolling: touch` on the viewport div. Unfortunately, putting the viewport div above the rows div breaks selection.

**Comment by tbodt** [comment]
Actually, it turns out that setting the z-index is unnecessary, and not setting the z-index makes selection work. So this is the perfect solution.

**Comment by Tyriar** [comment]
@tbodt https://github.com/xtermjs/xterm.js/issues/594 was kept open since ballistic scroll doesn't work yet.

**Pull request merged**